circuits/cpp/src/aztec3/circuits/kernel/public/common.cpp

```diff
@@ -37,6 +37,13 @@ void validate_this_public_call_hash(DummyComposer& composer,
                                     PublicKernelInputs<NT> const& public_kernel_inputs,
                                     KernelCircuitPublicInputs<NT>& public_inputs)
 {
+    // If public call stack is empty, we bail so array_pop doesn't throw_or_abort
+    if (array_length(public_inputs.end.public_call_stack) == 0) {
```

Comment:
I had considered putting this check in, but figured the error thrown by the pop would suffice. Was this not the case?

Reply:
The problem is that instead of returning to the caller with the serialized error, the function just crashed, since it called a `std::abort`.

Reply:
Got it.